plugins/tests/pod/test_pod.py

```diff
@@ -72,7 +73,9 @@ def test_dynamic_pod_task():
     def t1(a: int) -> int:
         return a + 10
 
-    @dynamic(task_config=dynamic_pod)
+    @dynamic(
+        task_config=dynamic_pod, requests=Resources(cpu="10"), limits=Resources(gpu="2"), environment={"FOO": "bar"}
+    )
     def dynamic_pod_task(a: int) -> List[int]:
```

Comment:
how do you make sure that the `dynamic behavior` is preserved in the unit test. For example, we should test for the dynamic spec right?

Reply:
we do! further below https://github.com/flyteorg/flytekit/pull/389/files#diff-7cda79b401da24c4efa25cf3db6dea45212c5fc87d370968c8254554a8721742R115